### PR TITLE
[FEAT]: Optimize Docker

### PR DESCRIPTION
**Pull request by ImBIOS**
## 📈 What's Improved?

image

- Use `clippy` instead of `clippy-preview`, it's more stable and works properly for this project.
- Add the `--no-cache` flag to the npm install command to avoid storing the package cache in the Docker layer.
- Cleans up the package cache after installing with apt-get.
- Use `postgres:12-alpine` instead of `postgres:12` to save up to 200 MB of image size
- Add comments with notes and todo to explicitly tell contributors about the current condition and flow of the code
- Add ImBIOS and use Dynamic Profile Image URL to automatically sync profile image with current contributor profile image (@griffobeid @darioalessandro I recommend this automated alternative: https://allcontributors.org/)
- Use vscode's `.devcontainer` feature for better developer experience and local-first development.
- Ignore unused dir into `.dockerignore`
- Compile `tailwindcss` for production in `trunk` hook
- From ~4 GB of total containers, reduced to ~1GB of total container size (-75%)
- Format `README.md` with `markdownlint`

## ✅ Objective

```[tasklist]
### Mandatory
- [x] Able to build production docker containers with multi-stage build technique
- [x] Production docker containers with multi-stage build technique running properly
- [x] Able to build development docker containers
- [x] Development docker containers running properly
- [x] Document on how to run production containers
- [x] Document on how to run development containers
```

```[tasklist]
### Optional
- [x] Able to run Dev Container
```

### CODE REVIEW

**Comment by ImBIOS** [comment]
The first step of #68

**Review by griffobeid — APPROVED** [review]
Thank you @ImBIOS !

**Comment by ImBIOS** [comment]
@griffobeid @darioalessandro I think it's ready for review now.

**Comment by ImBIOS** [comment]
@griffobeid I love your idea ⭐

I just need to sync and cook that idea, my current progress, and the current `main` branch containerization progress.

It'll be great developer experience to have a maintainable source code 👍

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
```
Comment (darioalessandro) [inline .devcontainer/devcontainer.json:1]: @ImBIOS is this for vscode?

File: Makefile [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+		docker compose -f docker/docker-compose.prod.yml run postgres bash -c "psql -h postgres -d actix-api-db -U postgres"
 
 clippy-fix:
-		docker compose -f docker/docker-compose.yaml run yew-ui bash -c "cd /app/yew-ui && cargo clippy --fix"
-		docker compose -f docker/docker-compose.yaml run actix-api bash -c "cd /app/actix-api && cargo clippy --fix"
+		docker compose -f docker/docker-compose.prod.yml run yew-ui bash -c "cd /app/yew-ui && cargo clippy --fix"
+		docker compose -f docker/docker-compose.prod.yml run actix-api bash -c "cd /app/actix-api && cargo clippy --fix"
 
 fmt:
-		docker compose -f docker/docker-compose.yaml run yew-ui bash -c "cd /app/yew-ui && cargo fmt"
```
Comment (darioalessandro) [inline Makefile:20]: @ImBIOS can we use the dev containers here?

File: .devcontainer/Dockerfile.actix.dev [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Builder and Runtime
```
Comment (darioalessandro) [inline .devcontainer/Dockerfile.actix.dev:1]: are these files ever used?

**Comment by griffobeid** [comment]
Hey @ImBIOS are you still working on this?

**Pull request closed without merging**